js/renovation/component_wrapper/utils/update_props_immutable.ts

```diff
@@ -1,10 +1,38 @@
-import { isPlainObject } from '../../../core/utils/type';
+import { isPlainObject, isString } from '../../../core/utils/type';
+
+function parseOptionName(fullName: string): (string | number)[] | null {
+  const parts: (string | number)[] = [];
+
+  fullName.split('.').forEach((part) => {
+    const match = /(.+?)\[\s*(\d+)\s*\]/g.exec(part);
+
+    if (match) {
+      parts.push(match[1], parseInt(match[2], 10));
+    } else {
+      parts.push(part);
+    }
+  });
+
+  // if any of parts contains ']' or '[' even after parsing, then it's incorrect
+  const isIncorrect = parts.some((part) => {
```

Comment:
Where is a test for incorrect prop name?

Reply:
There is test named `wrong array index`, this check is required only because of that test